{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cube Custom R Package\n",
    "This is the notebook to develop cube custom package.  All the package is in \"/home/jupypter/cubeR\" directory, and pushed to git\n",
    "https://github.com/TheJacksonLaboratory/cubeR\n",
    "for release.\n",
    "\n",
    "The package follow the directory structure by these tutorals:\n",
    "* http://r-pkgs.had.co.nz/intro.html\n",
    "* https://hilaryparker.com/2014/04/29/writing-an-r-package-from-scratch/\n",
    "* https://tinyheero.github.io/jekyll/update/2015/07/26/making-your-first-R-package.html\n",
    "\n",
    "To release\n",
    "* run \n",
    "```\n",
    "setwd(\"./cubeR\")\n",
    "document()\n",
    "```\n",
    "* git push to repo\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "install.packages(\"devtools\")\n",
    "library(\"devtools\")\n",
    "devtools::install_github(\"klutometis/roxygen\")\n",
    "library(roxygen2)\n",
    "\n",
    "setwd(\"/home/jupyter/cubeR\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[32m✔\u001b[39m Setting active project to \u001b[34m'/home/jupyter/cubeR/cubeR'\u001b[39m\n",
      "\n",
      "\u001b[32m✔\u001b[39m Saving \u001b[34m'snp_data'\u001b[39m to \u001b[34m'data/snp_data.rda'\u001b[39m\n",
      "\n",
      "\u001b[31m●\u001b[39m Document your data (see \u001b[34m'https://r-pkgs.org/data.html'\u001b[39m)\n",
      "\n",
      "\u001b[32m✔\u001b[39m Saving \u001b[34m'assay'\u001b[39m to \u001b[34m'data/assay.rda'\u001b[39m\n",
      "\n",
      "\u001b[31m●\u001b[39m Document your data (see \u001b[34m'https://r-pkgs.org/data.html'\u001b[39m)\n",
      "\n",
      "\u001b[32m✔\u001b[39m Saving \u001b[34m'disease'\u001b[39m to \u001b[34m'data/disease.rda'\u001b[39m\n",
      "\n",
      "\u001b[31m●\u001b[39m Document your data (see \u001b[34m'https://r-pkgs.org/data.html'\u001b[39m)\n",
      "\n",
      "\u001b[32m✔\u001b[39m Saving \u001b[34m'drug_treatment'\u001b[39m to \u001b[34m'data/drug_treatment.rda'\u001b[39m\n",
      "\n",
      "\u001b[31m●\u001b[39m Document your data (see \u001b[34m'https://r-pkgs.org/data.html'\u001b[39m)\n",
      "\n",
      "\u001b[32m✔\u001b[39m Saving \u001b[34m'mouse_strain'\u001b[39m to \u001b[34m'data/mouse_strain.rda'\u001b[39m\n",
      "\n",
      "\u001b[31m●\u001b[39m Document your data (see \u001b[34m'https://r-pkgs.org/data.html'\u001b[39m)\n",
      "\n",
      "Updating cubeR documentation\n",
      "\n",
      "Loading cubeR\n",
      "\n",
      "Warning message:\n",
      "“[/home/jupyter/cubeR/cubeR/R/hello.R:4] @param requires name and description”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing NAMESPACE\n",
      "Writing NAMESPACE\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[32m✔\u001b[39m Leaving \u001b[34m'vignettes/introduction.Rmd'\u001b[39m unchanged\n",
      "\n"
     ]
    }
   ],
   "source": [
    "setwd(\"/home/jupyter/cubeR/cubeR\")\n",
    "# need to figure out auto build \n",
    "snp_data <- read.csv(\"data-raw/snap_grid_sample.csv\")\n",
    "usethis::use_data(snp_data, overwrite = TRUE)\n",
    "\n",
    "assay <- read.csv(\"data-raw/assay.csv\")\n",
    "usethis::use_data(assay, overwrite = TRUE)\n",
    "\n",
    "disease <- read.csv(\"data-raw/disease.csv\")\n",
    "usethis::use_data(disease, overwrite = TRUE)\n",
    "\n",
    "drug_treatment <- read.csv(\"data-raw/drug_treatment.csv\")\n",
    "usethis::use_data(drug_treatment, overwrite = TRUE)\n",
    "\n",
    "mouse_strain <- read.csv(\"data-raw/mouse_strain.csv\")\n",
    "usethis::use_data(mouse_strain, overwrite = TRUE)\n",
    "\n",
    "document()\n",
    "usethis::use_vignette(\"introduction\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\u001b[32m✔\u001b[39m  \u001b[38;5;247mchecking for file ‘/home/jupyter/cubeR/cubeR/DESCRIPTION’\u001b[39m\u001b[36m\u001b[39m\n",
      "\u001b[38;5;247m─\u001b[39m\u001b[38;5;247m  \u001b[39m\u001b[38;5;247mpreparing ‘cubeR’:\u001b[39m\u001b[36m\u001b[39m\n",
      "\u001b[32m✔\u001b[39m  \u001b[38;5;247mchecking DESCRIPTION meta-information\u001b[39m\u001b[36m\u001b[39m\n",
      "\u001b[38;5;247m─\u001b[39m\u001b[38;5;247m  \u001b[39m\u001b[38;5;247mchecking for LF line-endings in source and make files and shell scripts\u001b[39m\u001b[36m\u001b[39m\n",
      "\u001b[38;5;247m─\u001b[39m\u001b[38;5;247m  \u001b[39m\u001b[38;5;247mchecking for empty or unneeded directories\u001b[39m\u001b[36m\u001b[39m\n",
      "\u001b[38;5;247m─\u001b[39m\u001b[38;5;247m  \u001b[39m\u001b[38;5;247mlooking to see if a ‘data/datalist’ file should be added\u001b[39m\u001b[36m\u001b[39m\n",
      "\u001b[38;5;247m─\u001b[39m\u001b[38;5;247m  \u001b[39m\u001b[38;5;247mbuilding ‘cubeR_0.0.0.9000.tar.gz’\u001b[39m\u001b[36m\u001b[39m\n",
      "   \n",
      "Running /usr/lib/R/bin/R CMD INSTALL /tmp/RtmpIHujGE/cubeR_0.0.0.9000.tar.gz \\\n",
      "  --install-tests \n",
      "\u001b[31m* installing to library ‘/usr/local/lib/R/site-library’\n",
      "\u001b[39m\u001b[31m* installing *source* package ‘cubeR’ ...\n",
      "\u001b[39m\u001b[31m** using staged installation\n",
      "\u001b[39m\u001b[31m** R\n",
      "\u001b[39m\u001b[31m** data\n",
      "\u001b[39m\u001b[31m*** moving datasets to lazyload DB\n",
      "\u001b[39m\u001b[31m** inst\n",
      "\u001b[39m\u001b[31m** byte-compile and prepare package for lazy loading\n",
      "\u001b[39m\u001b[31m** help\n",
      "\u001b[39m\u001b[31m*** installing help indices\n",
      "\u001b[39m\u001b[31m** building package indices\n",
      "\u001b[39m\u001b[31m** installing vignettes\n",
      "\u001b[39m\u001b[31m** testing if installed package can be loaded from temporary location\n",
      "\u001b[39m\u001b[31m** testing if installed package can be loaded from final location\n",
      "\u001b[39m\u001b[31m** testing if installed package keeps a record of temporary installation path\n",
      "\u001b[39m\u001b[31m* DONE (cubeR)\n",
      "\u001b[39m"
     ]
    }
   ],
   "source": [
    "setwd(\"/home/jupyter/cubeR\")\n",
    "install(\"cubeR\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"dataset:  dd\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 34 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>accession_id</th><th scope=col>subclass</th><th scope=col>target</th><th scope=col>seqclass</th><th scope=col>antibodies</th><th scope=col>short</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>JAXAS00001</td><td>bead array </td><td>naked gDNA  </td><td>N/A                    </td><td>no       </td><td>MUGA       </td></tr>\n",
       "\t<tr><td>JAXAS00002</td><td>sequencing </td><td>chromatin   </td><td>Open Chromatin         </td><td>no       </td><td>mATAC-Seq  </td></tr>\n",
       "\t<tr><td>JAXAS00003</td><td>anatomical </td><td>organism    </td><td>N/A                    </td><td>no       </td><td>Weight     </td></tr>\n",
       "\t<tr><td>JAXAS00004</td><td>behavioural</td><td>organism    </td><td>N/A                    </td><td>no       </td><td>Food       </td></tr>\n",
       "\t<tr><td>JAXAS00005</td><td>behavioural</td><td>organism    </td><td>N/A                    </td><td>no       </td><td>Water      </td></tr>\n",
       "\t<tr><td>JAXAS00006</td><td>metabolic  </td><td>metabolite  </td><td>N/A                    </td><td>no       </td><td>GTT        </td></tr>\n",
       "\t<tr><td>JAXAS00007</td><td>anatomical </td><td>organism    </td><td>N/A                    </td><td>no       </td><td>NMR        </td></tr>\n",
       "\t<tr><td>JAXAS00008</td><td>sequencing </td><td>bead array  </td><td>naked RNA              </td><td>RNA Seq  </td><td>no         </td></tr>\n",
       "\t<tr><td>JAXAS00009</td><td>metabolic  </td><td>            </td><td>                       </td><td>         </td><td>DEXA       </td></tr>\n",
       "\t<tr><td>JAXAS00010</td><td>metabolic  </td><td>            </td><td>                       </td><td>         </td><td>iCAL       </td></tr>\n",
       "\t<tr><td>JAXAS00011</td><td>microscopy </td><td>tissue      </td><td>N/A                    </td><td>yes      </td><td>Neuro      </td></tr>\n",
       "\t<tr><td>JAXAS00012</td><td>microscopy </td><td>tissue      </td><td>N/A                    </td><td>sometimes</td><td>Hist       </td></tr>\n",
       "\t<tr><td>JAXAS00013</td><td>sequencing </td><td>naked RNA   </td><td>cDNA/EST               </td><td>no       </td><td>t-Iso-Seq  </td></tr>\n",
       "\t<tr><td>JAXAS00014</td><td>sequencing </td><td>naked RNA   </td><td>cDNA/EST               </td><td>no       </td><td>Iso-Seq    </td></tr>\n",
       "\t<tr><td>JAXAS00015</td><td>sequencing </td><td>chromatin   </td><td>Open Chromatin         </td><td>no       </td><td>scATAC     </td></tr>\n",
       "\t<tr><td>JAXAS00016</td><td>sequencing </td><td>naked RNA   </td><td>RNA Seq                </td><td>no       </td><td>scRNA      </td></tr>\n",
       "\t<tr><td>JAXAS00017</td><td>sequencing </td><td>naked gDNA  </td><td>gDNA Seq               </td><td>no       </td><td>microWGS   </td></tr>\n",
       "\t<tr><td>JAXAS00018</td><td>sequencing </td><td>naked gDNA  </td><td>gDNA Seq               </td><td>no       </td><td>SV         </td></tr>\n",
       "\t<tr><td>JAXAS00019</td><td>sequencing </td><td>naked gDNA  </td><td>gDNA Seq               </td><td>no       </td><td>pWGS       </td></tr>\n",
       "\t<tr><td>JAXAS00020</td><td>sequencing </td><td>chromatin   </td><td>Chromosome Conformation</td><td>         </td><td>ChIA-Drop  </td></tr>\n",
       "\t<tr><td>JAXAS00021</td><td>sequencing </td><td>chromatin   </td><td>Chromosome Conformation</td><td>yes      </td><td>ChIA-PET   </td></tr>\n",
       "\t<tr><td>JAXAS00022</td><td>sequencing </td><td>PCR products</td><td>gDNA Seq               </td><td>no       </td><td>I6S        </td></tr>\n",
       "\t<tr><td>JAXAS00023</td><td>sequencing </td><td>PCR products</td><td>gDNA Seq               </td><td>no       </td><td>ITS        </td></tr>\n",
       "\t<tr><td>JAXAS00024</td><td>sequencing </td><td>naked RNA   </td><td>RNA Seq                </td><td>no       </td><td>DGE        </td></tr>\n",
       "\t<tr><td>JAXAS00025</td><td>sequencing </td><td>naked gDNA  </td><td>gDNA Seq               </td><td>no       </td><td>iWGS       </td></tr>\n",
       "\t<tr><td>JAXAS00026</td><td>sequencing </td><td>naked RNA   </td><td>RNA Seq                </td><td>no       </td><td>RNA-Seq    </td></tr>\n",
       "\t<tr><td>JAXAS00027</td><td>sequencing </td><td>chromatin   </td><td>Chromosome Conformation</td><td>no       </td><td>pcHiC      </td></tr>\n",
       "\t<tr><td>JAXAS00028</td><td>sequencing </td><td>protein     </td><td>ChIP-Seq               </td><td>yes      </td><td>ChIP-Seq   </td></tr>\n",
       "\t<tr><td>JAXAS00029</td><td>sequencing </td><td>chromatin   </td><td>Chromosome Conformation</td><td>         </td><td>HiC        </td></tr>\n",
       "\t<tr><td>JAXAS00030</td><td>sequencing </td><td>chromatin   </td><td>Open Chromatin         </td><td>no       </td><td>ATAC-Seq   </td></tr>\n",
       "\t<tr><td>JAXAS00031</td><td>bead array </td><td>naked gDNA  </td><td>N/A                    </td><td>no       </td><td>GenoArry   </td></tr>\n",
       "\t<tr><td>JAXAS00032</td><td>sequencing </td><td>naked RNA   </td><td>microRNA               </td><td>no       </td><td>miRNA      </td></tr>\n",
       "\t<tr><td>JAXAS00033</td><td>sequencing </td><td>naked gDNA  </td><td>gDNA Seq               </td><td>         </td><td>WGBS       </td></tr>\n",
       "\t<tr><td>JAXAS00034</td><td>bead array </td><td>naked gDNA  </td><td>gDNA Seq               </td><td>         </td><td>MethylArray</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 34 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " accession\\_id & subclass & target & seqclass & antibodies & short\\\\\n",
       " <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t JAXAS00001 & bead array  & naked gDNA   & N/A                     & no        & MUGA       \\\\\n",
       "\t JAXAS00002 & sequencing  & chromatin    & Open Chromatin          & no        & mATAC-Seq  \\\\\n",
       "\t JAXAS00003 & anatomical  & organism     & N/A                     & no        & Weight     \\\\\n",
       "\t JAXAS00004 & behavioural & organism     & N/A                     & no        & Food       \\\\\n",
       "\t JAXAS00005 & behavioural & organism     & N/A                     & no        & Water      \\\\\n",
       "\t JAXAS00006 & metabolic   & metabolite   & N/A                     & no        & GTT        \\\\\n",
       "\t JAXAS00007 & anatomical  & organism     & N/A                     & no        & NMR        \\\\\n",
       "\t JAXAS00008 & sequencing  & bead array   & naked RNA               & RNA Seq   & no         \\\\\n",
       "\t JAXAS00009 & metabolic   &              &                         &           & DEXA       \\\\\n",
       "\t JAXAS00010 & metabolic   &              &                         &           & iCAL       \\\\\n",
       "\t JAXAS00011 & microscopy  & tissue       & N/A                     & yes       & Neuro      \\\\\n",
       "\t JAXAS00012 & microscopy  & tissue       & N/A                     & sometimes & Hist       \\\\\n",
       "\t JAXAS00013 & sequencing  & naked RNA    & cDNA/EST                & no        & t-Iso-Seq  \\\\\n",
       "\t JAXAS00014 & sequencing  & naked RNA    & cDNA/EST                & no        & Iso-Seq    \\\\\n",
       "\t JAXAS00015 & sequencing  & chromatin    & Open Chromatin          & no        & scATAC     \\\\\n",
       "\t JAXAS00016 & sequencing  & naked RNA    & RNA Seq                 & no        & scRNA      \\\\\n",
       "\t JAXAS00017 & sequencing  & naked gDNA   & gDNA Seq                & no        & microWGS   \\\\\n",
       "\t JAXAS00018 & sequencing  & naked gDNA   & gDNA Seq                & no        & SV         \\\\\n",
       "\t JAXAS00019 & sequencing  & naked gDNA   & gDNA Seq                & no        & pWGS       \\\\\n",
       "\t JAXAS00020 & sequencing  & chromatin    & Chromosome Conformation &           & ChIA-Drop  \\\\\n",
       "\t JAXAS00021 & sequencing  & chromatin    & Chromosome Conformation & yes       & ChIA-PET   \\\\\n",
       "\t JAXAS00022 & sequencing  & PCR products & gDNA Seq                & no        & I6S        \\\\\n",
       "\t JAXAS00023 & sequencing  & PCR products & gDNA Seq                & no        & ITS        \\\\\n",
       "\t JAXAS00024 & sequencing  & naked RNA    & RNA Seq                 & no        & DGE        \\\\\n",
       "\t JAXAS00025 & sequencing  & naked gDNA   & gDNA Seq                & no        & iWGS       \\\\\n",
       "\t JAXAS00026 & sequencing  & naked RNA    & RNA Seq                 & no        & RNA-Seq    \\\\\n",
       "\t JAXAS00027 & sequencing  & chromatin    & Chromosome Conformation & no        & pcHiC      \\\\\n",
       "\t JAXAS00028 & sequencing  & protein      & ChIP-Seq                & yes       & ChIP-Seq   \\\\\n",
       "\t JAXAS00029 & sequencing  & chromatin    & Chromosome Conformation &           & HiC        \\\\\n",
       "\t JAXAS00030 & sequencing  & chromatin    & Open Chromatin          & no        & ATAC-Seq   \\\\\n",
       "\t JAXAS00031 & bead array  & naked gDNA   & N/A                     & no        & GenoArry   \\\\\n",
       "\t JAXAS00032 & sequencing  & naked RNA    & microRNA                & no        & miRNA      \\\\\n",
       "\t JAXAS00033 & sequencing  & naked gDNA   & gDNA Seq                &           & WGBS       \\\\\n",
       "\t JAXAS00034 & bead array  & naked gDNA   & gDNA Seq                &           & MethylArray\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 34 × 6\n",
       "\n",
       "| accession_id &lt;fct&gt; | subclass &lt;fct&gt; | target &lt;fct&gt; | seqclass &lt;fct&gt; | antibodies &lt;fct&gt; | short &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| JAXAS00001 | bead array  | naked gDNA   | N/A                     | no        | MUGA        |\n",
       "| JAXAS00002 | sequencing  | chromatin    | Open Chromatin          | no        | mATAC-Seq   |\n",
       "| JAXAS00003 | anatomical  | organism     | N/A                     | no        | Weight      |\n",
       "| JAXAS00004 | behavioural | organism     | N/A                     | no        | Food        |\n",
       "| JAXAS00005 | behavioural | organism     | N/A                     | no        | Water       |\n",
       "| JAXAS00006 | metabolic   | metabolite   | N/A                     | no        | GTT         |\n",
       "| JAXAS00007 | anatomical  | organism     | N/A                     | no        | NMR         |\n",
       "| JAXAS00008 | sequencing  | bead array   | naked RNA               | RNA Seq   | no          |\n",
       "| JAXAS00009 | metabolic   | <!----> | <!----> | <!----> | DEXA        |\n",
       "| JAXAS00010 | metabolic   | <!----> | <!----> | <!----> | iCAL        |\n",
       "| JAXAS00011 | microscopy  | tissue       | N/A                     | yes       | Neuro       |\n",
       "| JAXAS00012 | microscopy  | tissue       | N/A                     | sometimes | Hist        |\n",
       "| JAXAS00013 | sequencing  | naked RNA    | cDNA/EST                | no        | t-Iso-Seq   |\n",
       "| JAXAS00014 | sequencing  | naked RNA    | cDNA/EST                | no        | Iso-Seq     |\n",
       "| JAXAS00015 | sequencing  | chromatin    | Open Chromatin          | no        | scATAC      |\n",
       "| JAXAS00016 | sequencing  | naked RNA    | RNA Seq                 | no        | scRNA       |\n",
       "| JAXAS00017 | sequencing  | naked gDNA   | gDNA Seq                | no        | microWGS    |\n",
       "| JAXAS00018 | sequencing  | naked gDNA   | gDNA Seq                | no        | SV          |\n",
       "| JAXAS00019 | sequencing  | naked gDNA   | gDNA Seq                | no        | pWGS        |\n",
       "| JAXAS00020 | sequencing  | chromatin    | Chromosome Conformation | <!----> | ChIA-Drop   |\n",
       "| JAXAS00021 | sequencing  | chromatin    | Chromosome Conformation | yes       | ChIA-PET    |\n",
       "| JAXAS00022 | sequencing  | PCR products | gDNA Seq                | no        | I6S         |\n",
       "| JAXAS00023 | sequencing  | PCR products | gDNA Seq                | no        | ITS         |\n",
       "| JAXAS00024 | sequencing  | naked RNA    | RNA Seq                 | no        | DGE         |\n",
       "| JAXAS00025 | sequencing  | naked gDNA   | gDNA Seq                | no        | iWGS        |\n",
       "| JAXAS00026 | sequencing  | naked RNA    | RNA Seq                 | no        | RNA-Seq     |\n",
       "| JAXAS00027 | sequencing  | chromatin    | Chromosome Conformation | no        | pcHiC       |\n",
       "| JAXAS00028 | sequencing  | protein      | ChIP-Seq                | yes       | ChIP-Seq    |\n",
       "| JAXAS00029 | sequencing  | chromatin    | Chromosome Conformation | <!----> | HiC         |\n",
       "| JAXAS00030 | sequencing  | chromatin    | Open Chromatin          | no        | ATAC-Seq    |\n",
       "| JAXAS00031 | bead array  | naked gDNA   | N/A                     | no        | GenoArry    |\n",
       "| JAXAS00032 | sequencing  | naked RNA    | microRNA                | no        | miRNA       |\n",
       "| JAXAS00033 | sequencing  | naked gDNA   | gDNA Seq                | <!----> | WGBS        |\n",
       "| JAXAS00034 | bead array  | naked gDNA   | gDNA Seq                | <!----> | MethylArray |\n",
       "\n"
      ],
      "text/plain": [
       "   accession_id subclass    target       seqclass                antibodies\n",
       "1  JAXAS00001   bead array  naked gDNA   N/A                     no        \n",
       "2  JAXAS00002   sequencing  chromatin    Open Chromatin          no        \n",
       "3  JAXAS00003   anatomical  organism     N/A                     no        \n",
       "4  JAXAS00004   behavioural organism     N/A                     no        \n",
       "5  JAXAS00005   behavioural organism     N/A                     no        \n",
       "6  JAXAS00006   metabolic   metabolite   N/A                     no        \n",
       "7  JAXAS00007   anatomical  organism     N/A                     no        \n",
       "8  JAXAS00008   sequencing  bead array   naked RNA               RNA Seq   \n",
       "9  JAXAS00009   metabolic                                                  \n",
       "10 JAXAS00010   metabolic                                                  \n",
       "11 JAXAS00011   microscopy  tissue       N/A                     yes       \n",
       "12 JAXAS00012   microscopy  tissue       N/A                     sometimes \n",
       "13 JAXAS00013   sequencing  naked RNA    cDNA/EST                no        \n",
       "14 JAXAS00014   sequencing  naked RNA    cDNA/EST                no        \n",
       "15 JAXAS00015   sequencing  chromatin    Open Chromatin          no        \n",
       "16 JAXAS00016   sequencing  naked RNA    RNA Seq                 no        \n",
       "17 JAXAS00017   sequencing  naked gDNA   gDNA Seq                no        \n",
       "18 JAXAS00018   sequencing  naked gDNA   gDNA Seq                no        \n",
       "19 JAXAS00019   sequencing  naked gDNA   gDNA Seq                no        \n",
       "20 JAXAS00020   sequencing  chromatin    Chromosome Conformation           \n",
       "21 JAXAS00021   sequencing  chromatin    Chromosome Conformation yes       \n",
       "22 JAXAS00022   sequencing  PCR products gDNA Seq                no        \n",
       "23 JAXAS00023   sequencing  PCR products gDNA Seq                no        \n",
       "24 JAXAS00024   sequencing  naked RNA    RNA Seq                 no        \n",
       "25 JAXAS00025   sequencing  naked gDNA   gDNA Seq                no        \n",
       "26 JAXAS00026   sequencing  naked RNA    RNA Seq                 no        \n",
       "27 JAXAS00027   sequencing  chromatin    Chromosome Conformation no        \n",
       "28 JAXAS00028   sequencing  protein      ChIP-Seq                yes       \n",
       "29 JAXAS00029   sequencing  chromatin    Chromosome Conformation           \n",
       "30 JAXAS00030   sequencing  chromatin    Open Chromatin          no        \n",
       "31 JAXAS00031   bead array  naked gDNA   N/A                     no        \n",
       "32 JAXAS00032   sequencing  naked RNA    microRNA                no        \n",
       "33 JAXAS00033   sequencing  naked gDNA   gDNA Seq                          \n",
       "34 JAXAS00034   bead array  naked gDNA   gDNA Seq                          \n",
       "   short      \n",
       "1  MUGA       \n",
       "2  mATAC-Seq  \n",
       "3  Weight     \n",
       "4  Food       \n",
       "5  Water      \n",
       "6  GTT        \n",
       "7  NMR        \n",
       "8  no         \n",
       "9  DEXA       \n",
       "10 iCAL       \n",
       "11 Neuro      \n",
       "12 Hist       \n",
       "13 t-Iso-Seq  \n",
       "14 Iso-Seq    \n",
       "15 scATAC     \n",
       "16 scRNA      \n",
       "17 microWGS   \n",
       "18 SV         \n",
       "19 pWGS       \n",
       "20 ChIA-Drop  \n",
       "21 ChIA-PET   \n",
       "22 I6S        \n",
       "23 ITS        \n",
       "24 DGE        \n",
       "25 iWGS       \n",
       "26 RNA-Seq    \n",
       "27 pcHiC      \n",
       "28 ChIP-Seq   \n",
       "29 HiC        \n",
       "30 ATAC-Seq   \n",
       "31 GenoArry   \n",
       "32 miRNA      \n",
       "33 WGBS       \n",
       "34 MethylArray"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(cubeR)\n",
    "cubeR('Assay')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "head(assay, 6)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(ncol(snp_data))\n",
    "print(nrow(snp_data))\n",
    "\n",
    "head(metadata-assay, 6)\n",
    "# lapply(snp_data, mean, na.rm = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "row <- subset(snp_data, start_position=='110002640' && strain_name == \"129P2/OlaHsd:129P2/OlaHsd\" )\n",
    "print(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
